server/ast/expr.go

```diff
@@ -462,25 +492,42 @@ func nodeExpr(node tree.Expr) (vitess.Expr, error) {
 			Children:   vitess.Exprs{vitess.InjectedExpr{Expression: expr}},
 		}, nil
 	case *tree.DJSON:
-		return nil, fmt.Errorf("the statement is not yet supported")
+		// TODO: should we use json.JSON for JSON type values?
```

Comment:
My first thought here is that the JSON might not be validated, and that this should be a text conversion instead (which would catch malformed JSON errors). We definitely need tests for how Postgres handles such cases.

I think this should be a string, not the `json.JSON` type. Internally, IIRC, Postgres uses string manipulation on the JSON type.